src/main/java/org/opentripplanner/routing/algorithm/raptor/transit/request/RaptorRequestTransferCache.java

```diff
@@ -0,0 +1,152 @@
+package org.opentripplanner.routing.algorithm.raptor.transit.request;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+import lombok.EqualsAndHashCode;
+import lombok.SneakyThrows;
+import org.opentripplanner.routing.algorithm.raptor.transit.Transfer;
+import org.opentripplanner.routing.algorithm.raptor.transit.TransitLayer;
+import org.opentripplanner.routing.api.request.RoutingRequest;
+import org.opentripplanner.routing.api.request.StreetMode;
+import org.opentripplanner.routing.core.BicycleOptimizeType;
+import org.opentripplanner.transit.raptor.api.transit.RaptorTransfer;
+
+public class RaptorRequestTransferCache {
+
+    private static final LoadingCache<CacheKey, List<List<RaptorTransfer>>> transferCache = CacheBuilder
+        .newBuilder()
+        .maximumSize(25)
```

Comment:
Isn't 25 quite low?

Reply:
I think this is an entire set of transfers for a given configuration of routing options, so 25 should be plenty.

Reply:
There is around 20 fields in the key and the number of combinations will vary a lot from deployment to deployment, if no restrictions is put on the client this will easily go towards ~infinite combinations. Setting this will be a "tuning job" to keep it below acceptable memory usage and catching the most commonly used profiles.

Reply:
Should the value be configurable using `router-config.json`?

Reply:
I've added a commit so that the value may be configured using `router-config.json`:

```json
  "transit": {
    "transferCacheMaxSize": 10
  }
```